finetune/util/table_labeler.py

```diff
@@ -677,11 +678,54 @@ def __init__(
             split_preds_to_cells=split_preds_to_cells,
             chunk_tables=chunk_tables,
         )
-        self.table_model = SequenceLabeler(
-            base_model=TableRoBERTa, **(table_model_config or {})
+        # Delay construction of these models.
+        self._get_table_model = functools.partial(
+            SequenceLabeler, base_model=TableRoBERTa, **(table_model_config or {})
         )
-        self.text_model = SequenceLabeler(**(text_model_config or {}))
-        self.table_chunker = TableChunker.from_table_model(self.table_model)
+        self._get_text_model = functools.partial(
+            SequenceLabeler, **(text_model_config or {})
+        )
+
+        self.temp_dir = tempfile.TemporaryDirectory()
+        self.text_model_path = os.path.join(self.temp_dir.name, "text.jl")
+        self.table_model_path = os.path.join(self.temp_dir.name, "table.jl")
+
+    @property
+    @contextlib.contextmanager
+    def text_model(self):
+        if os.path.exists(self.text_model_path):
+            text_model = SequenceLabeler.load(self.text_model_path)
+        else:
+            text_model = self._get_text_model()
+        yield text_model
+        text_model.save(self.text_model_path)
+
+    @property
+    @contextlib.contextmanager
```

Comment:
Do we need the context manager? As long as we handle all of this within the _fit_text_model function it should go out of scope when the _fit_text_model fn terminates and get cleaned up? NBD obviously, would just be a bit easier to follow flow of code.